pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
> No, that's exactly what I am not talking about. I want accurate sums over long time periods (like total logins per hour/day/week). There should not be a need to do any average in that calculation.

We think of that in terms of average logins per second.

> Did not know about the increase function. Would be interested to know how it works 

It's syntactic sugar over `rate`, it multiplies the rate by the time range.

> Not seeing any significant inaccuracies here. 

Presuming all the data gets there.